ocs_ci/deployment/tnf.py

```
@@ -0,0 +1,543 @@
+        """
+        proxy_config = self.tnf_config.get("proxy", {})
+        if not proxy_config.get("enabled", True):
+            return
+
+        cluster_path = config.ENV_DATA.get("cluster_path", "")
+        if cluster_path and self.hypervisor.load_instance_info(cluster_path):
+            proxy_url = self.hypervisor.get_proxy_url()
+            os.environ["HTTPS_PROXY"] = proxy_url
+            os.environ["HTTP_PROXY"] = proxy_url
+            os.environ["NO_PROXY"] = "localhost,127.0.0.1,.svc,.cluster.local"
+            logger.info(f"Proxy configured from metadata: {proxy_url}")
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)